lib/connectors/memory.js

```diff
@@ -351,26 +351,54 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+  function cast(prop, val, props) {
+    if (Array.isArray(val)) {
+      return val.map(function(val) {
+        return cast(prop, val, props);
+      });
+    }
+
+    var isArray = Array.isArray(props[prop].type);
+    var name = isArray ? props[prop].type[0].name : props[prop].type.name;
+
+    switch (name) {
+      case 'Date':
+        val = new Date(val.toString().replace(/GMT.*$/, 'GMT'));
+        break;
+      case 'Boolean':
+        val = Boolean(val);
+        break;
+      case 'Number':
+        val = Number(val);
+        break;
+      case 'ModelConstructor':
+        for (var subProp in val) {
```

Comment:
I still don't think this is a good idea. Could you please show me the use case where it's needed to convert plain-data objects into Model instances?